debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -408,12 +414,19 @@ protected void handleGtidEvent(Event event) {
         String gtid = gtidEvent.getGtid();
         gtidSet.add(gtid);
         source.startGtid(gtid, gtidSet.toString()); // rather than use the client's GTID set
+        ignoreDmlEventByGtidSource = false;
+        if (gtidDmlSourceFilter != null && gtid != null) {
+            String uuid = gtid.substring(0, gtid.indexOf(":"));
```

Comment:
Is this going to work if GTIDs have more than one UUID in them? Not quite sure how DML GTIDs look--do they only ever have one UUID? It would make sense if so, but just want to confirm.